test/extensions/stats_sinks/hystrix/hystrix_test.cc

```diff
@@ -0,0 +1,370 @@
+
+  // set gauge value.
+  ON_CALL(cluster2_stats_scope, gauge("membership_total"))
+      .WillByDefault(ReturnRef(membership_total_gauge_));
+
+  // attach counters
+  NiceMock<Stats::MockCounter> success_counter2;
+  NiceMock<Stats::MockCounter> error_5xx_counter2;
+  NiceMock<Stats::MockCounter> retry_5xx_counter2;
+  NiceMock<Stats::MockCounter> error_4xx_counter2;
+  NiceMock<Stats::MockCounter> retry_4xx_counter2;
+  success_counter2.name_ = "upstream_rq_2xx";
+  ON_CALL(cluster2_stats_scope, counter("upstream_rq_2xx"))
+      .WillByDefault(ReturnRef(success_counter2));
+  error_5xx_counter2.name_ = "upstream_rq_5xx";
+  ON_CALL(cluster2_stats_scope, counter("upstream_rq_5xx"))
+      .WillByDefault(ReturnRef(error_5xx_counter2));
+  retry_5xx_counter2.name_ = "retry.upstream_rq_5xx";
+  ON_CALL(cluster2_stats_scope, counter("retry.upstream_rq_5xx"))
+      .WillByDefault(ReturnRef(retry_5xx_counter2));
+  error_4xx_counter2.name_ = "upstream_rq_4xx";
+  ON_CALL(cluster2_stats_scope, counter("upstream_rq_4xx"))
+      .WillByDefault(ReturnRef(error_4xx_counter2));
+  retry_4xx_counter2.name_ = "retry.upstream_rq_4xx";
+  ON_CALL(cluster2_stats_scope, counter("retry.upstream_rq_4xx"))
+      .WillByDefault(ReturnRef(retry_4xx_counter2));
+
+  cluster2_info->stats().upstream_rq_timeout_.reset();
+  cluster2_info->stats().upstream_rq_per_try_timeout_.reset();
+  cluster2_info->stats().upstream_rq_pending_overflow_.reset();
+
+  // add cluster
+  cluster_map_.emplace(cluster2_name, cluster2);
+  Buffer::OwnedImpl buffer = createClusterAndCallbacks();
+
+  // generate data to both clusters
+  for (int i = 0; i < 12; i++) {
+    buffer.drain(buffer.length());
+    // cluster 1
+    ON_CALL(error_5xx_counter_, value()).WillByDefault(Return((i + 1) * 17));
+    ON_CALL(success_counter_, value()).WillByDefault(Return((i + 1) * 7));
+    cluster1_info_->stats().upstream_rq_timeout_.add(1);
+    cluster1_info_->stats().upstream_rq_per_try_timeout_.add(2);
+    cluster1_info_->stats().upstream_rq_pending_overflow_.add(8);
+
+    // cluster 2
+    ON_CALL(error_5xx_counter2, value()).WillByDefault(Return((i + 1) * 1));
+    ON_CALL(retry_5xx_counter2, value()).WillByDefault(Return((i + 1) * 2));
+    ON_CALL(error_4xx_counter2, value()).WillByDefault(Return((i + 1) * 3));
+    ON_CALL(retry_4xx_counter2, value()).WillByDefault(Return((i + 1) * 4));
+    ON_CALL(success_counter2, value()).WillByDefault(Return((i + 1) * 3));
+    cluster2_info->stats().upstream_rq_timeout_.add(3);
+    cluster2_info->stats().upstream_rq_per_try_timeout_.add(1);
+    cluster2_info->stats().upstream_rq_pending_overflow_.add(5);
+
+    sink_->flush(source_);
+  }
+
+  std::string data_message = TestUtility::bufferToString(buffer);
+
+  std::size_t pos1 = data_message.find(cluster1_name_);
+  std::size_t pos2 = data_message.find(cluster2_name);
+  EXPECT_NE(std::string::npos, pos1);
+  EXPECT_NE(std::string::npos, pos2);
```

Comment:
probably want to make these ASSERT_NE because you don't want to use relational operators below on pos1 or pos2 if they are npos.